recipe/meta.yaml

```diff
@@ -45,35 +44,37 @@ requirements:
     - python
     - pip
     - cython
-    - arrow-cpp >=10.0.1,<11
-    - pyarrow
-    - numpy
-    - setuptools >=40.6.0
+    - arrow-cpp 10.0.1
+    - pyarrow  10.0.1
+    - numpy  {{ numpy }}
+    - setuptools
     - wheel
   run:
     - python
     - packaging
-    - pyarrow
+    - pyarrow  >=10.0.1,<10.1.0
     - {{ pin_compatible('arrow-cpp', max_pin='x') }}
     - asn1crypto >0.24.0,<2.0.0
     - cffi >=1.9,<2.0.0
-    - cryptography >=3.1.0,<41.0.0
+    - cryptography >=3.1.0,<42.0.0
     - oscrypto <2.0.0
     - pyopenssl >=16.2.0,<24.0.0
     - pycryptodomex >=3.2,!=3.5.0,<4.0.0
     - pyjwt <3.0.0
     - pytz
     - requests <3.0.0
-    - importlib-metadata  # [py<38]
-    - charset-normalizer >=2,<3
+    - charset-normalizer >=2,<4
     - idna >=2.5,<4
     - urllib3 >=1.21.1,<1.27
     - certifi >=2017.4.17
     - typing-extensions >=4.3.0,<5
     - filelock >=3.5,<4
+    - sortedcontainers >=2.4.0
+    - platformdirs >=2.6.0,<3.9.0
+    - tomlkit
   run_constrained:
-    - pandas >1.0.0,<1.6.0
-    - keyring !=16.1.0,<24.0.0
+    - pandas >1.0.0,<2.1.0
+    - keyring !=16.1.0,<25.0.0
 
```

Comment:
I think `numpy<1.26.0` is also an optional dependency. If you look at how it's used [here](https://github.com/snowflakedb/snowflake-connector-python/blob/1380e41cd2088f0c83af0cf45010d68aac96b366/src/snowflake/connector/arrow_context.py#L25), this means numpy may have runtime restrictions, should you choose to run the optional code.
Right now our `cbc.yaml` does not conflict, but it may in the future.

Reply:
numpy (optional) is mentioned as a development dependency.

Reply:
I thought we would use this section also for development dependencies.

Reply:
let's leave it out for now since it's an optional dependency and a clear policy is not in place.